tests/conftest.py

```diff
@@ -472,21 +472,21 @@ def _decode_token_by_position_index(
 
     def generate_greedy_logprobs_nm_use_tokens(
         self,
-        prompts: List[str],
+        input_ids_lst: List[torch.Tensor],
```

Comment:
* previously, we were passing a prompt, formatted with a chat template (which appends bos token)
* then, we tokenize here which appends another bos token

This change prevents that double <bos> permanently, by forcing the test to tokenize everything fully